assign_rights/tests.py

```diff
@@ -95,21 +109,60 @@ def test_grouping_form(self):
 class TestRightsAssembler(TestCase):
     def setUp(self):
         add_rights_shells()
+        add_rights_acts()
         self.assembler = RightsAssembler()
+        self.rights_ids = [obj.pk for obj in RightsShell.objects.all()]
 
     def test_retrieve_rights(self):
         """Tests the retrieve_rights method.
 
         Asserts the method returns a list or DoesNotExist exception.
         """
-        rights_ids = [obj.pk for obj in RightsShell.objects.all()]
-        assembled = self.assembler.retrieve_rights(rights_ids)
+        assembled = self.assembler.retrieve_rights(self.rights_ids)
         self.assertTrue(isinstance(assembled, list))
-        self.assertEqual(len(rights_ids), len(assembled))
+        self.assertEqual(len(self.rights_ids), len(assembled))
 
-        rights_ids.append(len(rights_ids) + 1)
+        deleted = random.choice(RightsShell.objects.all())
+        deleted.delete()
+        self.rights_ids.append(deleted.pk)
         with self.assertRaises(RightsShell.DoesNotExist):
-            assembled = self.assembler.retrieve_rights(rights_ids)
+            assembled = self.assembler.retrieve_rights(self.rights_ids)
+
+    def test_calculate_dates(self):
+        """Tests the calculate_dates method.
+
+        Asserts the method returns a DateTimeField or None if the end date is open.
+        Asserts that the relative delta of the calculated date and the correct end date
+        is equal to the end date period of the object.
+        """
+        assembled = self.assembler.retrieve_rights(self.rights_ids)
```

Comment:
I would suggest just using a Django queryset `RightsShell.objects.all()` that way you keep your tests isolated and a failure in `test_retrieve_rights` won't cause a failure in `test_calculate_dates`.